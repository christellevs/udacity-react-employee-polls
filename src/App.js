import "./App.css";
import { useEffect, Fragment } from "react";
import { connect } from "react-redux";
import { Route, Routes } from "react-router-dom";
import { handleInitialData } from "./actions/shared.js";
import Navbar from "./components/Navbar.js";
import Dashboard from "./components/Dashboard.js";
import Leaderboard from "./components/Leaderboard.js";
import PollPage from "./components/PollPage.js";
import NewPoll from "./components/NewPoll.js";
import Login from "./components/Login.js";
import LoadingBar from "react-redux-loading-bar";

function App(props) {
  useEffect(() => {
    props.dispatch(handleInitialData());
  }, []);

  return (
    <div className="App">
      <LoadingBar />
      <div>
        <Navbar />
        {props.loading === true ? null : (
          <Routes>
            <Route exact path="/" element={<Dashboard />} />
            <Route exact path="/add" element={<NewPoll />} />
            <Route exact path="/leaderboard" element={<Leaderboard />} />
            <Route exact path="/questions/:id" element={<PollPage />} />
            <Route exact path="/login" element={<Login />} />
          </Routes>
        )}
      </div>
    </div>
  );
}

const mapStateToProps = ({ authedUser }) => ({ loading: authedUser === null });

export default connect(mapStateToProps)(App);
