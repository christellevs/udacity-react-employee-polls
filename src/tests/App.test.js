import * as React from "react";
import { render } from "@testing-library/react";
import App from "../App.js";
import { Provider } from "react-redux";
import { store } from "../store.js";
import { BrowserRouter } from "react-router-dom";

test("renders app", () => {
  var component = render(
    <Provider store={store}>
      <BrowserRouter>
        <App />
      </BrowserRouter>
    </Provider>
  );
  expect(component).toMatchSnapshot();
});
