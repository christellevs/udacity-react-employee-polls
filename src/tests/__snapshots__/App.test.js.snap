// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders app 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="App"
      >
        <div
          style="direction: ltr;"
        >
          <div
            class=""
            style="width: 0%; transition: width 800ms linear 0s; will-change: width, opacity; height: 3px; background-color: red; position: absolute; opacity: 0;"
          />
          <div
            style="display: table; clear: both;"
          />
        </div>
        <div>
          <div
            class="login-container"
          >
            <h2>
              Employee Web Polls
            </h2>
            <hr />
            <form>
              <div>
                <h3>
                  Please select a user:
                </h3>
                <select
                  class="login-input"
                >
                  <option
                    disabled=""
                    selected=""
                    value="none"
                  >
                    Select user...
                  </option>
                </select>
              </div>
              <button
                class="btn btn-login"
                disabled=""
                type="submit"
              >
                Login
              </button>
            </form>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="App"
    >
      <div
        style="direction: ltr;"
      >
        <div
          class=""
          style="width: 0%; transition: width 800ms linear 0s; will-change: width, opacity; height: 3px; background-color: red; position: absolute; opacity: 0;"
        />
        <div
          style="display: table; clear: both;"
        />
      </div>
      <div>
        <div
          class="login-container"
        >
          <h2>
            Employee Web Polls
          </h2>
          <hr />
          <form>
            <div>
              <h3>
                Please select a user:
              </h3>
              <select
                class="login-input"
              >
                <option
                  disabled=""
                  selected=""
                  value="none"
                >
                  Select user...
                </option>
              </select>
            </div>
            <button
              class="btn btn-login"
              disabled=""
              type="submit"
            >
              Login
            </button>
          </form>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
